### PR TITLE
pg improvements & tests

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@seut can you review this? I think there is nothing blocking this from being merged.

File: sql/build.gradle [hunk 1]
```diff
@@ -20,7 +20,9 @@ dependencies {
     testCompile project(':testing')
     testCompile 'org.skyscreamer:jsonassert:1.2.0'
     testCompile 'org.hamcrest:hamcrest-all:1.3'
-    testCompile 'org.postgresql:postgresql:9.4.1209.jre7'
+    testCompile ('com.github.mfussenegger:pgjdbc-jre7:dev-SNAPSHOT') {
```
Comment (mfussenegger) [inline sql/build.gradle]: Should I add a comment here indicating why this forked version is necessary?

Comment (seut) [inline sql/build.gradle]: yes pls, wanted to ask just now ;)

File: sql/src/main/java/io/crate/protocols/postgres/types/PGArray.java [hunk 2]
```diff
@@ -68,49 +71,125 @@ public Object readBinaryValue(ChannelBuffer buffer, int valueLength) {
...
+    }
+
+    // Decode individual inner object
+    private void addObject(byte[] bytes, int startIdx, int endIdx, List<Object> objects) {
+        if (endIdx > startIdx) {
+            byte firstValueByte = bytes[startIdx];
+            if (firstValueByte == '"') {
+                byte[] innerBytes = Arrays.copyOfRange(bytes, startIdx + 1, endIdx);
+                String s = new String(innerBytes, StandardCharsets.UTF_8);
+                s = s.replace("\\\"", "\"");
```
Comment (seut) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGArray.java]: I think we should here use custom replacing logic instead, like we do for json arrays for better performance.
At least let's add an todo, and do this asap in a different PR.

File: sql/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 3]
```diff
@@ -647,13 +652,15 @@ public void testCreateTableWithCustomSchema() throws Exception {
     }
 
     @Test
+    @UseJdbc(false) // estoCrateException rewrite is missing
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java]: should be solved now or?

File: sql/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java [hunk 4]
```diff
@@ -124,6 +126,7 @@ public void testInWithArgs() throws Exception {
     }
 
     @Test
+    @UseJdbc(false) // geo_shape missing
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java]: isn't geo_shape support already in?

File: sql/src/test/java/io/crate/integrationtests/RepositoryIntegrationTest.java [hunk 5]
```diff
@@ -61,6 +64,7 @@ public void testDropExistingRepository() throws Exception {
     }
 
     @Test
+    @UseJdbc(false) // result contains bytesref instead of string
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/RepositoryIntegrationTest.java]: should be solved or?

File: sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 6]
```diff
@@ -1512,6 +1515,7 @@ public void testGeoTypeQueries() throws Exception {
     }
 
     @Test
+    @UseJdbc(false) // shape support is missing
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java]: already solved due to geo_shape support?

**Comment by seut** [comment]
left some comments

**Comment by seut** [comment]
as discussed pls check the geo_shape array commit ;)

**Comment by seut** [comment]
lgtm
